Yoast/wordpress-seo
# Fix fatal in term builder

## Context


* When a term doesn't exists and a primary term records has been set a fatal error occurs. We don't want that to happen.

## Summary


This PR can be summarized in the following changelog entry:

* N/A - Fixes a bug where a fatal error is given when a term doesn't exist.

## Relevant technical choices:

* Chosen to do a `get_term` check. Also improved the term builder with throwing an exception when term isn't found (`null`). 
* Decided to not write tests because this is a private method, the class is complex and it would take a lot of time to test this. 

## Test instructions

This PR can be tested by following these steps:

* On the indexables branch
* Create a post and attach some terms to it. Mark a term as primary one.
* Remove all records in the tables: indexables and indexables_hierarchy
* Remove the term that is set as primary term from the table `wp_terms` 
* Visit a post and see an fatal error is given
* Checkout this branch and when needed repeat all steps
* The fatal error should be gone.

## UI changes

## Documentation

## Quality assurance


Fixes #14631
